File: node/paginate.mjs
const API_URL = "https://api.compsniper.com/v1/scrape";
const apiKey = process.env.COMPSNIPER_API_KEY?.trim();
if (!apiKey) throw new Error("Set COMPSNIPER_API_KEY before running this example.");

const keyword = process.env.COMPSNIPER_KEYWORD ?? "sony wh-1000xm5";
const maxPages = Math.max(1, Math.min(Number(process.env.COMPSNIPER_MAX_PAGES ?? 3), 100));
const items = [];

for (let page = 1; page <= maxPages; page += 1) {
  const url = new URL(API_URL);
  url.searchParams.set("keyword", keyword);
  url.searchParams.set("page", String(page));
  url.searchParams.set("count", "240");

  const response = await fetch(url, {
    headers: { Authorization: `Bearer ${apiKey}` },
    signal: AbortSignal.timeout(75_000),
  });
  const data = await response.json();
  if (!response.ok) throw new Error(data.error ?? `CompSniper returned HTTP ${response.status}`);

  items.push(...data.items);
  console.log(`Page ${page}: ${data.totalItems} listings`);
  if (!data.hasNextPage) break;
}

console.log(`Collected ${items.length} listing rows across at most ${maxPages} pages.`);

File: python/paginate.py
"""Fetch a bounded number of result pages. Each successful page uses one request."""

from __future__ import annotations

import os

import requests

API_URL = "https://api.compsniper.com/v1/scrape"


def required_api_key() -> str:
    key = os.environ.get("COMPSNIPER_API_KEY", "").strip()
    if not key:
        raise SystemExit("Set COMPSNIPER_API_KEY before running this example.")
    return key


keyword = os.environ.get("COMPSNIPER_KEYWORD", "sony wh-1000xm5")
max_pages = max(1, min(int(os.environ.get("COMPSNIPER_MAX_PAGES", "3")), 100))
items: list[dict] = []

for page in range(1, max_pages + 1):
    response = requests.get(
        API_URL,
        headers={"Authorization": f"Bearer {required_api_key()}"},
        params={"keyword": keyword, "page": page, "count": 240},
        timeout=75,
    )
    response.raise_for_status()
    data = response.json()
    items.extend(data.get("items", []))
    print(f"Page {page}: {data.get('totalItems', 0)} listings")
    if not data.get("hasNextPage"):
        break

print(f"Collected {len(items)} listing rows across at most {max_pages} pages.")
